tracker_test.go

```
@@ -925,3 +927,176 @@ func TestRun_Result_BundleIdentity_EmptyWhenNotSet(t *testing.T) {
+	if err == nil {
+		t.Fatalf("expected preflight failure on non-repo workdir with --git=require")
+	}
+	if !errors.Is(err, pipeline.ErrGitWorkdirNotRepo) {
+		t.Errorf("want ErrGitWorkdirNotRepo, got %v", err)
+	}
+}
+
+func TestNewEngine_PreflightPassesAfterGitInit(t *testing.T) {
+	requireGit(t)
+	dir := t.TempDir()
+	cmd := exec.Command("git", "init", "-q")
```

golangci-lint (2.12.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)